codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexCodec.java

```diff
@@ -149,40 +149,34 @@ final void onHttp2Frame(ChannelHandlerContext ctx, Http2Frame frame) {
         ctx.fireChannelRead(frame);
     }
 
-    private void onHttp2UpgradeStreamInitialized(ChannelHandlerContext ctx, DefaultHttp2FrameStream stream) {
-        assert stream.state() == Http2Stream.State.HALF_CLOSED_LOCAL;
-        AbstractHttp2StreamChannel ch = new Http2MultiplexCodecStreamChannel(stream, null);
-        ch.closeOutbound();
-
-        // Add our upgrade handler to the channel and then register the channel.
-        // The register call fires the channelActive, etc.
-        ch.pipeline().addLast(upgradeStreamHandler);
-        ChannelFuture future = ctx.channel().eventLoop().register(ch);
-        if (future.isDone()) {
-            Http2MultiplexHandler.registerDone(future);
-        } else {
-            future.addListener(Http2MultiplexHandler.CHILD_CHANNEL_REGISTRATION_LISTENER);
-        }
-    }
-
     @Override
     final void onHttp2StreamStateChanged(ChannelHandlerContext ctx, DefaultHttp2FrameStream stream) {
-
         switch (stream.state()) {
```

Comment:
This looks identical to the Http2MultiplexHandler code other than the type of `AbstractHttp2StreamChannel` created (`Http2MultiplexCodecStreamChannel` vs `Http2MultiplexHandlerStreamChannel`). Could that be lifted into `Http2ChannelDuplexHandler`?

Reply:
Maybe it's okay since the multiplex codec is deprecated and presumably will be removed at some point?

Reply:
yeah that was my thinking